src/mapillary.py

```diff
@@ -0,0 +1,145 @@
+import json
+import logging
+from multiprocessing import Pool
+from pathlib import Path
+from typing import Annotated, Optional
+
+from requests import RequestException, Session
+from requests.adapters import HTTPAdapter
+from shapely import Point
+from typer import Argument, Option, Typer
+
+from src.data_parsing.geopandas import GeoPandasParser
+
+log = logging.getLogger(__name__)
+log.setLevel(logging.INFO)
+log.addHandler(logging.StreamHandler())
+app = Typer()
+
+
+class Mapillary:
+    url = "https://graph.mapillary.com/images"
+
+    def __init__(
+        self,
+        access_token,
+        basepath=Path(Path(__file__).parent.parent, "data/raw/mapillary"),
+    ):
+        self.access_token = access_token
+        self.basepath = basepath
+        self.basepath.mkdir(parents=True, exist_ok=True)
+        self.client = Session()
+        self.client.mount("https://", HTTPAdapter(max_retries=3))
```

Comment:
For retries, I recommend using a package like [stamina](https://github.com/hynek/stamina) which gives you a nice decorator and has nicer default behavior like exponential backoff. 

Then you could just do something like

```python
@stamina.retry(on=requests.HTTPError, attempts=3)
def get_image_from_coordinates(...)
```

and 

```python
@stamina.retry(on=requests.HTTPError, attempts=3)
def _download_image(...)
```

(I'm not sure what the errors you're getting are, so `requests.HTTPError` may not be the exact thing to put there.)

Reply:
okay if i capture this an issue for future consideration?

Reply:
Gonna leave this in a future issue for now, I'm getting multiple errors and can't catch them all, and also stamina only seems to be moderately popular/has few issues activity and I don't want to depend on an optional package until we do some more investigation on the best path forward.

Reply:
logged as #24 to revisit later